model/src/test/kotlin/config/ExcludesTest.kt

```diff
@@ -0,0 +1,207 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model.config
+
+import com.here.ort.model.AnalyzerResult
+import com.here.ort.model.Identifier
+import com.here.ort.model.PackageReference
+import com.here.ort.model.Project
+import com.here.ort.model.Scope
+
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.WordSpec
+
+class ExcludesTest : WordSpec() {
+    init {
+        "findProjectExclude" should {
+            val project = Project.EMPTY.copy(definitionFilePath = "path2")
+
+            "return null if there is no matching project exclude" {
+                val excludes = Excludes(
+                        projects = listOf(
+                                ProjectExclude("path1", ProjectExcludeReason.BUILD_TOOL_OF, comment = "comment"),
+                                ProjectExclude("path3", ProjectExcludeReason.BUILD_TOOL_OF, comment = "comment")
+                        )
+                )
+
+                excludes.findProjectExclude(project) shouldBe null
+            }
+
+            "find the correct project exclude" {
+                val projectExclude = ProjectExclude("path2", ProjectExcludeReason.BUILD_TOOL_OF, comment = "comment")
+
+                val excludes = Excludes(
+                        projects = listOf(
+                                ProjectExclude("path1", ProjectExcludeReason.BUILD_TOOL_OF, comment = "comment"),
+                                projectExclude,
+                                ProjectExclude("path3", ProjectExcludeReason.BUILD_TOOL_OF, comment = "comment")
+                        )
+                )
+
+                excludes.findProjectExclude(project) shouldBe projectExclude
+            }
+        }
+
+        "isPackageExcluded" should {
+            val id = Identifier("provider", "namespace", "name", "version")
+            val projectId1 = Identifier("provider", "namespace", "project1", "version")
+            val projectId2 = Identifier("provider", "namespace", "project2", "version")
+
+            "return true if the package does not appear in the analyzer result" {
+                val excludes = Excludes()
+
+                val analyzerResult = AnalyzerResult(
+                        projects = sortedSetOf(),
+                        packages = sortedSetOf(),
+                        errors = sortedMapOf()
+                )
+
+                excludes.isPackageExcluded(id, analyzerResult) shouldBe true
+            }
+
+            "return true if all occurrences of the package are excluded" {
+                val excludes = Excludes(
+                        projects = listOf(ProjectExclude("path1", ProjectExcludeReason.BUILD_TOOL_OF, "comment")),
```

Comment:
In order to remove a bit redundancy and make the code more readable please extract a
function and re-use it from all the tests, e.g.
projectExcludesOfPaths( vararg excludePaths: String )